spec/factories/actions.rb

```diff
@@ -0,0 +1,13 @@
+FactoryBot.define do
```

Comment:
FactoryBot is primarily used in tests. It defines 'factories' that know how to build our models with sensible default values. This factory will allow us to write something like 'FactoryBot.create(:action), which would create an Action with name = "MyString", cost = 1, etc. It would also save it to the database. 'FactoryBot.build(:action)' would create it without saving it. You can overide the default values too, for example: 'FactoryBot.create(:action, name: 'My Custom Action)' I haven't filled these out with sensible defaults yet, obviously, this is basically what was autogenerated for me, when I generated the models.

Reply:
Additional notes on FactoryBot:
You aren't limited to one factory per class, typically you'll have a factory for each 'important state' an object might be in. For user, for example, you might have new_user and confirmed_user. Factories can also inherit from each other, so you can reuse the defaults for the fields a new_user has when creating confirmed_user.